tests/Agent/IntegrationTests/IntegrationTests/Logging/ContextDataNotSupportedTests.cs

```diff
@@ -0,0 +1,127 @@
+
+namespace NewRelic.Agent.IntegrationTests.Logging.ContextData
+{
+    public abstract class ContextDataNotSupportedTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly TFixture _fixture;
+        private LoggingFramework _loggingFramework;
+        private bool _contextDataEnabled;
+
+        private const string InfoMessage = "HelloWorld";
+
+        // There are several entries in this dictionary to allow for different methods of adding the values in the test adapter
+        // If you need more entries for your framework, add them
+        private Dictionary<string, string> _expectedAttributes = new Dictionary<string, string>()
+        {
+            { "mycontext1", "foo" },
+            { "mycontext2", "bar" },
+            { "mycontext3", "test" },
+            { "mycontext4", "value" },
+        };
+
+
+        public ContextDataNotSupportedTestsBase(TFixture fixture, ITestOutputHelper output, bool contextDataEnabled, LoggingFramework loggingFramework) : base(fixture)
+        {
+            _fixture = fixture;
+            _loggingFramework = loggingFramework;
+            _contextDataEnabled = contextDataEnabled;
+            _fixture.SetTimeout(TimeSpan.FromMinutes(2));
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand($"LoggingTester SetFramework {_loggingFramework}");
+            _fixture.AddCommand($"LoggingTester Configure");
+
+            string context = string.Join(",", _expectedAttributes.Select(x => x.Key + "=" + x.Value).ToArray());
+
+            // should generate an exception message in the log since the dummy ILogger doesn't have the right properties
+            _fixture.AddCommand($"LoggingTester CreateSingleLogMessage {InfoMessage} INFO {context}");
+            // do it again - this time, context data should be marked as unsupported and not generate an exception in the log
+            _fixture.AddCommand($"LoggingTester CreateSingleLogMessage {InfoMessage} INFO {context}");
+
+            _fixture.AddActions
+            (
+                setupConfiguration: () =>
+                {
+                    var configModifier = new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath);
+
+                    configModifier
+                    .EnableLogMetrics(true)
+                    .EnableLogForwarding(true)
+                    .EnableContextData(_contextDataEnabled)
+                    .EnableDistributedTrace()
+                    .SetLogLevel("finest");
+                },
+                exerciseApplication: () =>
+                {
+                    Thread.Sleep(TimeSpan.FromSeconds(2));
+                }
+            );
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var match = _fixture.AgentLog.TryGetLogLines(AgentLogBase.ContextDataNotSupportedLogLineRegex);
+            Assert.Single(match);
```

Comment:
Test needs to assert that logs were actually sent as well. Checking that we logged about context data being disabled is different that verifying we still forward logs.

Reply:
Good point. Updated to verify logs were forwarded but without any context data.